packages/api/src/auth.js

```diff
@@ -0,0 +1,121 @@
+import { gql } from '@web3-storage/db'
+import * as JWT from './utils/jwt.js'
+import {
+  UserNotFoundError,
+  TokenNotFoundError,
+  UnrecognisedTokenError,
+  NoTokenError,
+  MagicTokenRequiredError
+} from './errors.js'
+
+/**
+ * Middleware: verify the request is authenticated with a valid magic link token.
+ * On successful login, adds a `auth` property on the Request
+ *
+ * @param {import('itty-router').RouteHandler} handler
+ * @returns {import('itty-router').RouteHandler}
+ */
+export function withMagicToken (handler) {
+  /**
+   * @param {Request} request
+   * @param {import('./env').Env}
+   * @returns {Response}
+   */
+  return async (request, env, ctx) => {
+    const token = getTokenFromRequest(request, env)
+    if (isMagicToken(token, env)) {
```

Comment:
```suggestion
    if (await isMagicToken(token, env)) {
```

Reply:
almost! turns out `magic.token.validate` is sync, and will throw if it doesn't recognise it. I'm going to catch that error and return false.